progression/src/main/java/me/mykindos/betterpvp/progression/progression/perks/DropMultiplierFishingPerk.java

```diff
@@ -0,0 +1,89 @@
+package me.mykindos.betterpvp.progression.progression.perks;
+
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import me.mykindos.betterpvp.core.config.Config;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilServer;
+import me.mykindos.betterpvp.progression.Progression;
+import me.mykindos.betterpvp.progression.model.ProgressionPerk;
+import me.mykindos.betterpvp.progression.model.ProgressionTree;
+import me.mykindos.betterpvp.progression.model.stats.ProgressionData;
+import me.mykindos.betterpvp.progression.tree.fishing.Fishing;
+import me.mykindos.betterpvp.progression.tree.fishing.event.PlayerStopFishingEvent;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Item;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+
+import java.util.Objects;
+
+@BPvPListener
+@Singleton
+@Slf4j
+public class DropMultiplierFishingPerk implements Listener, ProgressionPerk, DropMultiplier {
+
+    @Config(path = "fishing.perks.drop-multiplier.minLevel", defaultValue = "0")
+    @Inject
+    int minLevel;
+
+    @Config(path = "fishing.perks.drop-multiplier.maxLevel", defaultValue = "1000")
+    @Inject
+    int maxLevel;
+
+    @Config(path = "fishing.perks.drop-multiplier.increasePerLevel", defaultValue = "0.25")
+    @Inject
+    double increasePerLevel;
+
+    @Inject
+    private Progression progression;
+
+    @Inject
+    private Fishing fishing;
+
+
+    @Override
+    public String getName() {
+        return "Drop Multiplier Fishing";
+    }
+
+    @Override
+    public Class<? extends ProgressionTree>[] acceptedTrees() {
+        return new Class[] {
+                Fishing.class
+        };
+    }
+
+    @Override
+    public boolean canUse(Player player, ProgressionData<?> data) {
+        return minLevel <= data.getLevel() && data.getLevel() <= maxLevel;
+    }
+
+    @EventHandler(priority = EventPriority.HIGH)
+    public void onCatch(PlayerStopFishingEvent event) {
```

Comment:
I'm going to be honest, idk why I didn't do it this way before. Probably because I was confused on how the iteraction between Catching a fish and overriding it was actually happening.